pkg/network/dns_parser.go

```diff
@@ -160,47 +160,37 @@ func (p *dnsParser) parseAnswerInto(
 		return nil
 	}
 
-	var alias []byte
-	domainQueried := question.Name
 	pktInfo.queryType = QueryType(question.Type)
-
-	// Retrieve the CNAME record, if available.
-	alias = p.extractCNAME(domainQueried, dns.Answers)
-	if alias == nil {
-		alias = p.extractCNAME(domainQueried, dns.Additionals)
-	}
-
-	// Get IPs
-	p.extractIPsInto(alias, domainQueried, dns.Answers, t)
-	p.extractIPsInto(alias, domainQueried, dns.Additionals, t)
-	t.dns = strings.ToLower(string(domainQueried))
+	alias := p.extractCNAME(question.Name, dns.Answers)
+	p.extractIPsInto(alias, dns.Answers, t)
+	t.dns = strings.ToLower(string(question.Name))
```

Comment:
I think we can cut off another allocation here by replacing this line by:
```
t.dns = string(bytes.ToLower(question.Name))
```

Reply:
👍